be/src/olap/rowset/segment_v2/binary_dict_page.cpp

```diff
@@ -102,7 +102,16 @@ Slice BinaryDictPageBuilder::finish() {
     Slice data_slice = _data_page_builder->finish();
     _buffer.append(data_slice.data, data_slice.size);
     encode_fixed32_le(&_buffer[0], _encoding_type);
-    return Slice(_buffer.data(), _buffer.size());
+
+    if (_encoding_type == DICT_ENCODING) {
+        size_t dict_offset = _buffer.size();
+        Slice dictionary_page;
+        get_dictionary_page(&dictionary_page);
```

Comment:
there may be memory leak. You should delete dictionary_page.data after usage

Reply:
👌